### PR TITLE
Remove usage of hardcoded link to latest execution version

### PR DESCRIPTION
**Pull request by dariorussi**
## Description 

Few crates were hardcoding references to the execution version which makes difficult to change the execution versioning with a `v_last_` as a production version and a `latest` as a working version.
This PR removes those dependencies.
The entire story for the verifier as a cli command is a bit questionable but I think it moves us forward so it should be good for now

## Test Plan 

Existing tests

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-types/src/executable_transaction.rs
Comment (dariorussi) [inline crates/sui-types/src/executable_transaction.rs:69]: is this a good home for those functions?
@lxfind @tnowacki @mystenmark @tzakian @amnn @bmwill

File: crates/sui/src/client_commands.rs
Comment (dariorussi) [inline crates/sui/src/client_commands.rs:778]: @oxade could you please review this part?
You'll notice I defined an entry point in the execution traits that deals with this specifically and I am not sure how critical. I think it'll be fine to keep that around so no big deal, but I wonder if you (or anybody else) think there is a good way to do this

File: sui-execution/src/verifier.rs
Comment (dariorussi) [inline sui-execution/src/verifier.rs:42]: this is the code added to allow changes in verification runs. 
We can expand `VerifierOverrides` in time as needed.
I think this is ok, but don't know if people have strong feelings about it

File: crates/sui-types/src/executable_transaction.rs
Comment (lxfind) [reply crates/sui-types/src/executable_transaction.rs:69]: Probably not. I think sui-types/execution.rs would be a better place.

File: sui-execution/src/v0.rs
Comment (amnn) [inline sui-execution/src/v0.rs:217]: The interface of mutating the overrides is a little funky -- could we not split overrides into a read-only part and a return part?

**Review by amnn — COMMENTED** [review]
Overall looks good, thanks @dariorussi (will wait to stamp because it looks like there are pending comments and questions for other reviewers but let me know if you want me to unblock).  The two remaining issues I see around introducing a new `v_last` are:

- how does the CLI know that it's running on devnet and therefore should have latest Vs running on mainnet/testnet and therefore is using some snapshot.
- testing (e.g. transactional testing) which I now realise uses the Authority so already uses execution versioning, but now it becomes even more important that we have a way to test changes to multiple execution versions (which we don't currently do).

File: sui-execution/Cargo.toml
Comment (dariorussi) [reply sui-execution/Cargo.toml:30]: well the "cutting a new execution layer" test is failing and hopefully this is the issue. I have not investigated too much yet but overall I think this broke that so a bit more work on this PR still.

File: sui-execution/src/v0.rs
Comment (dariorussi) [reply sui-execution/src/v0.rs:217]: we could and it would be a bit more plumbing. Let me try that and see how it looks.
To be honest I am not too clear we need this whole story and it was mostly to minimize changes.
This is just to get a specific command in the CLI working and don't know....

**Comment by dariorussi** [comment]
thanks @amnn for the comments and also take a look at the "cutting a new execution layer" failure in case there is some suggestions you have. I am sure I'll figure out the problem but your experience there is pretty invaluable.
For testing the current plan (and in the next PR) is to run **all** tests in both "devnet" and "mainnet" mode (in whichever form we can). That was our latest concern (and something you pointed out) and it was a valid and key concern. 
So the hope is that by running on both protocol versions we would get the cover we want.
I have not investigated the details of that yet and it is the next PR.
This PR seemed to have value by itself anyway so that is why it's here.
I'll address the issues and let you know if I need your "blessing" though you may be on PTO soon, so enjoy it!

File: crates/sui-types/src/executable_transaction.rs
Comment (dariorussi) [reply crates/sui-types/src/executable_transaction.rs:69]: will do, thanks!

**Comment by dariorussi** [comment]
> how does the CLI know that it's running on devnet and therefore should have latest Vs running on mainnet/testnet and therefore is using some snapshot.

good question and I am not sure. The idea is to play with the protocol config and some environment variable or similar. We'll see, hopefully it'll be straightforward.
If you have suggestions of course let me know

**Comment by dariorussi** [comment]
@lxfind and @amnn feedback

**Comment by dariorussi** [comment]
this is good to go now if anybody wanted to accept it....

File: sui-execution/Cargo.toml
Comment (amnn) [inline sui-execution/Cargo.toml:33]: :+1:

File: sui-execution/src/latest.rs
Comment (amnn) [inline sui-execution/src/latest.rs:202]: Don't you need to set these back to how they were, or, even better, seeing as you are passing a reference to the `config` in the call below, how about taking a copy of the `config` and editing that instead?

File: sui-execution/src/suivm.rs
Comment (amnn) [inline sui-execution/src/suivm.rs:202]: (ditto)

File: sui-execution/src/v0.rs
Comment (amnn) [inline sui-execution/src/v0.rs:208]: (ditto)

**Review by amnn — APPROVED** [review]
LGTM, but note the comments on resetting overrides on the verifier config! (And also we noted in an earlier comment somewhere that we need to come up with some reasonable way for users to experience the devnet version of the VM and verifier when running against `devnet` and the "production" version otherwise.

File: sui-execution/src/latest.rs
Comment (dariorussi) [reply sui-execution/src/latest.rs:202]: yes will do and good point!

**Comment by dariorussi** [comment]
> LGTM, but note the comments on resetting overrides on the verifier config! 

will fix before commit

**Comment by dariorussi** [comment]
feedback

**Pull request merged**
